app/Http/Controllers/PlaceController.php

```diff
@@ -45,25 +39,41 @@ public function index($id)
     }
 
     public function getRecommendPlace($id)
-    {       
+    {   
+        # place_keyword tableから$idとplace_idが等しいデータを抽出
         $place_keywords = DB::table('place_keywords')->where('place_id', $id)->get();
-        $tmp = [];
-
-        foreach($place_keywords as $place_keyword){
-                $tmp[] = $place_keyword->keyword_id;
-        }
-
-        $target = DB::table('place_keywords')->whereIn('keyword_id', $tmp)->where('place_id', '<>', $id)->get();
-
-        $places = [];
-        for($i = 1; $i < 5; $i++)
-        {
-            $random_num = rand(0, $target->count()-1);
-            $record = $target[$random_num];
-            $place = DB::table('places')->where('id', $record->place_id)->first();
-            $places[] =$place;
+        // dd(empty($place_keywords));
+        if(count($place_keywords) == 0){
+            $recommend_place = [];
         }
-        return $places;
+        else{
+            $keyword_id_list = [];
+            foreach($place_keywords as $place_keyword){
+                    $keyword_id_list[] = $place_keyword->keyword_id;
+            }
+    
+            $all_target_data = DB::table('place_keywords')->whereIn('keyword_id', $keyword_id_list)->where('place_id', '<>', $id)->get();
```

Comment:
please make use of the models and their relationships

Reply:
I think it is better to use the model when I select the data from the database but it doesn't work well if I wrote it in the model.

Reply:
It should work, we have to make use of models and relationships. There's no need to use raw query here